### ISSUE TITLE
CssRewrite goes through app_dev.php in Symfony

### ISSUE THREAD

**Issue by beberlei** [kriswallsmith/assetic#53]
Using symfony2 and a cssrewrite to fix all @import they now go through app_dev.php/path/to/import/file.css which leads to 404 errors from the symfony kernel.

**Comment by n1c01a5** [comment]
Same problem.

**Comment by matiaspub** [comment]
Same problem here with Symfony 2.6
Amazing!!! 4 yers!!!

**Comment by clubdesarrolladores** [comment]
+1 same problem, please fix this soon

**Comment by JHGitty** [comment]
I think there will be never a fix for this. The ``Best Practice`` currently is to use css files directory in the ``web`` directory (I prefer ``web/css`` and think this is the best way).

See this warning in the Symfony2 Book: http://symfony.com/doc/current/cookbook/assetic/asset_management.html#including-css-stylesheets

**Comment by drAlberT** [comment]
I can't see any "best practice" sentence on the doc .. simply it makes us aware of the issue

+1 for taking this issue on a solution

**Comment by JHGitty** [comment]
Here is the ``Best Practice`` to store the css files in the ``web`` directory:
http://symfony.com/doc/current/best_practices/web-assets.html

**Comment by drAlberT** [comment]
@JHGitty yep I see, but this is a bp for application .. if I have a bundle to redistribute or reuse I *need* to have my assets bundled together with the code who uses them 

So I agree to avoid misuse of bundled-assets, but they are a tool that has its own field of application and can't be ignored or labeled as "poor practice" always IMHO

However, I think assets should be treated all the same way ... so there should be no difference in the use of css, js or images .. I agree to use ` {% stylesheets 'bundles/app/css/*' filter='cssrewrite' %}`, but we should do the same with JS and images then :)

**Comment by stphane** [comment]
**app_dev** is driving me crazy with a web/fonts/... issue !
Giving this stylesheet snippet:
```
            {% stylesheets
                filter="node_scss,uglifycss"
                output="dist/main.css"
                '../app/Resources/public/scss/main.scss'
            %}
                <link rel="stylesheet" href="{{ asset_url }}">
            {% endstylesheets %}
```
The resulting css "/app_dev.php/dist/main_main_1.css" file points to some bootstrap glyphicons fonts referenced with "/fonts/bootstrap/glyphi..." okey so I copied this fonts directory to /web.

I then expected the browser to search for those files inside "/app_dev.php/fonts/bootstrap/glyph…", this is what is done but a 404 is served instead by the dev controller :angry: !
If I hit the url directly from vhost root "/fonts/bootstrap/glyph…", the file is correctly served by NGINX.

Is there a way to make app_dev resolve the **/fonts** path nicely (I mean in a clean way) ?
Could someone please help me handling this ?
Thank you !

**Comment by ReneMuetti** [comment]
This is a design bug. Since you do not work through a config switch, but with
different scripts that is an unavoidable consequence.

I had the same problem. The support was unfortunately no help. Also sont there
are no viable solutions. I therefore once assembled a "solution":
the page-template (base.html.twig)
`<!DOCTYPE html>



<html lang="de"> 
    <head>
        <meta charset="UTF-8" />
        <meta name="viewport" content="width=device-width, initial-scale=1, maximum-scale=1">
        {{ include('AppBundle:default:scss.html.twig') }}
....
`
then the CSS file (scss.html.twig)
`{% if app.environment == 'dev' %}
    {% stylesheets filter="scssphp" output="css/app_dev.css" debug=true
        "assets/scss/global-dev.scss"
        "assets/css/*.css"
    %}
    <link rel="stylesheet" type="text/css" href="{{ asset_url }}" media="screen">
    {% endstylesheets %}
{% else %}
    {% stylesheets filter="scssphp" output="css/app_prod.css" debug=true
        "assets/scss/global-prod.scss"
        "assets/css/*.css"
    %}
    <link rel="stylesheet" type="text/css" href="{{ asset_url }}" media="screen">
    {% endstylesheets %}
{% endif %}`

then generate 2 different scss:
one contains the paths for productive operation ( '../fonts/' ) and the other,
the paths for the DEV mode ( '../../fonts/' ).

Another solution I have not found. The Simple tile cause must be resolved by the
development team. Both app*.php together are placed and the DEV mode must be
regulated through a config option.

**Comment by stphane** [comment]
@ReneMuetti Thank you very much for sharing !
In the meantime, I ended up turning use_controller switch to false inside the configuration file and I now use assetic's watcher in dev environment.
I didn't think of wrapping Assetic tags between conditions since Assetic does not interpret twig files but only parses it in order to process stylesteets and javascripts tags. Actually it does make sense ! I will definitely concider your «solution» for futur projects.
